homeassistant/components/airly/air_quality.py

```diff
@@ -0,0 +1,207 @@
+
+_LOGGER = logging.getLogger(__name__)
+
+ATTRIBUTION = "Data provided by Airly"
+
+ATTR_CAQI = "CAQI"
+ATTR_CAQI_ADVICE = "advice"
+ATTR_CAQI_DESCRIPTION = "DESCRIPTION"
+ATTR_CAQI_LEVEL = "air quality index level"
+ATTR_PM_10_LIMIT = "PM10_LIMIT"
+ATTR_PM_10_PERCENT = "PM10_PERCENT"
+ATTR_PM_2_5_LIMIT = "PM25_LIMIT"
+ATTR_PM_2_5_PERCENT = "PM25_PERCENT"
+LABEL_PM_2_5_LIMIT = f"{ATTR_PM_2_5} limit"
+LABEL_PM_2_5_PERCENT = f"{ATTR_PM_2_5} percent of limit"
+LABEL_PM_10_LIMIT = f"{ATTR_PM_10} limit"
+LABEL_PM_10_PERCENT = f"{ATTR_PM_10} percent of limit"
+
+DEFAULT_SCAN_INTERVAL = timedelta(minutes=10)
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Add a Airly entities from a config_entry."""
+    api_key = config_entry.data[CONF_API_KEY]
+    name = config_entry.data[CONF_NAME]
+    latitude = config_entry.data[CONF_LATITUDE]
+    longitude = config_entry.data[CONF_LONGITUDE]
+    language = config_entry.data[CONF_LANGUAGE]
+    scan_interval = DEFAULT_SCAN_INTERVAL
+
+    websession = async_get_clientsession(hass)
+
+    data = AirlyData(
+        websession, api_key, latitude, longitude, language, scan_interval=scan_interval
+    )
+
+    await data.async_update()
+
+    async_add_entities([AirlyAirQuality(data, name)], True)
+
+
+def round_state(func):
+    """Round state."""
+
+    def _decorator(self):
+        res = func(self)
+        if isinstance(res, float):
+            return round(res)
+        return res
+
+    return _decorator
+
+
+class AirlyAirQuality(AirQualityEntity):
+    """Define an Airly air_quality."""
+
+    def __init__(self, airly, name):
+        """Initialize."""
+        self.airly = airly
+        self.data = airly.data
+        self._name = name
+        self._pm_2_5 = None
+        self._pm_10 = None
+        self._caqi = None
+        self._icon = "mdi:blur"
+        self._attrs = {}
+
+    @property
+    def name(self):
+        """Return the name."""
+        return self._name
+
+    @property
+    def icon(self):
+        """Return the icon."""
+        return self._icon
+
+    @property
+    @round_state
+    def air_quality_index(self):
+        """Return the air quality index."""
+        return self._caqi
+
+    @property
+    @round_state
+    def particulate_matter_2_5(self):
+        """Return the particulate matter 2.5 level."""
+        return self._pm_2_5
+
+    @property
+    @round_state
+    def particulate_matter_10(self):
+        """Return the particulate matter 10 level."""
+        return self._pm_10
+
+    @property
+    def attribution(self):
+        """Return the attribution."""
+        return ATTRIBUTION
+
+    @property
+    def state(self):
+        """Return the CAQI description."""
+        return self.data[ATTR_CAQI_DESCRIPTION]
+
+    @property
+    def unique_id(self):
+        """Return a unique_id for this entity."""
+        return f"{DOMAIN}-{self.airly.latitude}-{self.airly.longitude}"
```

Comment:
We don't need to add the domain. The entity registry is aware of the integration and platform when checking uniqueness.
```suggestion
        return f"{self.airly.latitude}-{self.airly.longitude}"
```